stdlib/public/Cxx/CxxConvertibleToCollection.swift

```diff
@@ -21,36 +21,36 @@ public protocol CxxConvertibleToCollection {
   mutating func __endUnsafe() -> RawIterator
 }
 
-@inlinable @inline(__always)
-internal func forEachElement<C: CxxConvertibleToCollection>(
-  of c: C,
-  body: (C.RawIterator.Pointee) -> Void
-) {
-  var mutableC = c
-  withExtendedLifetime(mutableC) {
-    var rawIterator = mutableC.__beginUnsafe()
-    let endIterator = mutableC.__endUnsafe()
-    while rawIterator != endIterator {
-      body(rawIterator.pointee)
-      rawIterator = rawIterator.successor()
+extension CxxConvertibleToCollection {
+  @inlinable
+  internal func forEach(_ body: (RawIterator.Pointee) -> Void) {
```

Comment:
What if this was made `public` and declared `rethrows`? 

(Other than the usual `forEach` annoyance about not being able to stop iterating without throwing an error.)

Reply:
I totally agree that `public` forEach could be convenient, but I think it is not strictly necessary for now. I think we can make this `public` a bit later once we see demand for it and settle on the public interface.